pkg/adapter/github/adapter.go

```diff
@@ -42,35 +40,28 @@ const (
 // Adapter converts incoming GitHub webhook events to CloudEvents and
 // then sends them to the specified Sink
 type Adapter struct {
-	Sink   string
-	client client.Client
+	SinkURI string
+	client  client.Client
 
 	initClientOnce sync.Once
 }
 
 // HandleEvent is invoked whenever an event comes in from GitHub
-func (ra *Adapter) HandleEvent(payload interface{}, header webhooks.Header) {
+func (a *Adapter) HandleEvent(payload interface{}, header webhooks.Header) {
 	hdr := http.Header(header)
-	err := ra.handleEvent(payload, hdr)
+	err := a.handleEvent(payload, hdr)
 	if err != nil {
 		log.Printf("unexpected error handling GitHub event: %s", err)
 	}
 }
 
-func (ra *Adapter) handleEvent(payload interface{}, hdr http.Header) error {
-	ra.initClientOnce.Do(func() {
-		var err error
-		if ra.client, err = client.NewHTTPClient(
-			client.WithTarget(ra.Sink),
-			client.WithHTTPBinaryEncoding(),
-			client.WithUUIDs(),
-			client.WithTimeNow(),
-		); err != nil {
-			log.Printf("failed to create cloudevent client: %s", err)
-		}
+func (a *Adapter) handleEvent(payload interface{}, hdr http.Header) error {
+	var err error
+	a.initClientOnce.Do(func() {
```

Comment:
this too please :)

Reply:
this one does not have a start method. 😕

Reply:
I think it would be better if this file would have a NewAdapter method and we'd do it there. But fine...